GroupBundle/EventListener/UpdateNodeGroupRoleListener.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+namespace OpenOrchestra\GroupBundle\EventListener;
+
+use Doctrine\ODM\MongoDB\Event\PreUpdateEventArgs;
+use OpenOrchestra\BackofficeBundle\Model\GroupInterface;
+use OpenOrchestra\BackofficeBundle\Model\NodeGroupRoleInterface;
+use OpenOrchestra\GroupBundle\Exception\NodeGroupRoleNotFoundException;
+use OpenOrchestra\ModelInterface\Model\NodeInterface;
+use OpenOrchestra\ModelInterface\Repository\NodeRepositoryInterface;
+
+/**
+ * Class UpdateNodeGroupRoleListener
+ */
+class UpdateNodeGroupRoleListener
+{
+    protected $nodeClass;
+
+    /**
+     * @param string $nodeClass
+     */
+    public function __construct($nodeClass)
+    {
+        $this->nodeClass = $nodeClass;
+    }
+
+    /**
+     * @param PreUpdateEventArgs $event
+     *
+     * @throws NodeGroupRoleNotFoundException
+     */
+    public function preUpdate(PreUpdateEventArgs $event)
+    {
+        $document = $event->getDocument();
+        $uow = $event->getDocumentManager()->getUnitOfWork();
+        if (
+            $document instanceof NodeGroupRoleInterface &&
+            $event->hasChangedField("accessType")
+        ) {
+            $parentAssociation = $uow->getParentAssociation($document);
```

Comment:
Yes maybe.

As a matter of fact, I think that it is not so great to rely our code on doctrine listeners because they are specific to doctrine odm's events.

@itkg-alavieille would it be possible to make the same modification in a listener listening on nodeGroupRole events ?